### PR TITLE
add user_manual(), add heart texture and solve the issue of update_ba…

### PR DESCRIPTION
**Pull request by JeanneXu1122**

### CODE REVIEW

File: src/generate_view.c [hunk 1]
```diff
@@ -52,6 +64,22 @@ void generate_view(SDL_Window* p_window, map_t* p_map)
     SDL_Rect player_rect = {  p_map->player.current_pos.x, p_map->player.current_pos.y,  w, h};
     SDL_RenderCopy(p_renderer, p_player_texture, NULL, &player_rect);
 
+    SDL_QueryTexture(p_heart_texture, NULL, NULL, &w, &h);
+    SDL_Rect heart_rect = { p_map->space.x_max-50, p_map->space.y_max,  w, h };
+    SDL_RenderCopy(p_renderer, p_heart_texture, NULL, &heart_rect);
+
+    itoa(p_map->player.heart, heartNum, 10);
```
Comment (Pari-Firoozi) [inline src/generate_view.c:71]: Unfortunately itoa is not an standard function! I don't have it in MAC check https://stackoverflow.com/questions/10162465/implicit-declaration-of-function-itoa-is-invalid-in-c99.

Can you please replace line 71 with this line:
sprintf(heartNum, "%d", p_map->player.heart);

it does exactly the same thing.

File: src/main.c [hunk 2]
```diff
@@ -79,7 +85,12 @@ int main()
                 }
                 else if (event.key.keysym.sym == SDLK_RETURN)
                 {
-                    if (menu.selector == MENU_ITEM_SINGLE_PLAYER)
+                    if (menu.selector == MENU_ITEM_USER_MANUAL)
+                    {
+                        quit = user_manual(p_window);
```
Comment (Pari-Firoozi) [inline src/main.c:90]: user_manual prototype is defined wrongly in include/menu.h can you please fix it as well. I couldn't compile it without fixing it.

File: src/generate_view.c [hunk 3]
```diff
@@ -5,24 +5,36 @@
 
 #include"../include/map.h"
 #include <stdio.h>
+
+#ifdef _WIN32
+	#include <SDL.h>
+#elif __APPLE__
 #include <SDL2/SDL.h>
 #include <SDL2/SDL_image.h>
```
Comment (Pari-Firoozi) [inline src/generate_view.c:13]: Add #include <SDL2/SDL_ttf.h> here.

**Review by Pari-Firoozi — COMMENTED** [review]
Thank you Jin. It works very good and the heart texture and position in the signle_player looks very professional and nice. 

Just can you please add src/user_manual.c item in the Makefile and some minor comments I added in the review.

File: src/generate_view.c (hunk 1 above)
Comment (JeanneXu1122) [reply src/generate_view.c:71]: ok

File: src/main.c (hunk 2 above)
Comment (JeanneXu1122) [reply src/main.c:90]: I don't get the problem you encountered, the project executes correctly on my end.

File: src/generate_view.c (hunk 3 above)
Comment (JeanneXu1122) [reply src/generate_view.c:13]: ok

**Review by Jaserelhabrouk — APPROVED** [review]
Looks good

**Review by Pari-Firoozi — APPROVED** [review]

**Review by layeghimahsa1996 — APPROVED** [review]
Looks so good Jin

**Pull request merged**